Source/Core/Core/Movie.cpp

```diff
@@ -854,9 +854,9 @@ void RecordInput(GCPadStatus* PadStatus, int controllerID)
 
   CheckPadStatus(PadStatus, controllerID);
 
-  EnsureTmpInputSize((size_t)(s_currentByte + 8));
-  memcpy(&(tmpInput[s_currentByte]), &s_padState, 8);
-  s_currentByte += 8;
+  EnsureTmpInputSize((size_t)(s_currentByte + sizeof(ControllerState)));
+  *reinterpret_cast<ControllerState*>(&tmpInput[s_currentByte]) = s_padState;
```

Comment:
> If a program attempts to access the stored value of an object through a glvalue of other than one of the following types. the behavior is undefined:
> - a char or unsigned char type.

Okay, so the way I'm reading that is, you can treat any pointer type as a character pointer, but not the other way around?